core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SparqlSession.java

```diff
@@ -172,6 +188,24 @@ public SparqlSession(HttpClient client, ExecutorService executor) {
 		// parser used for processing server response data should be lenient
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_DATATYPE_VALUES);
 		parserConfig.addNonFatalError(BasicParserSettings.VERIFY_LANGUAGE_TAGS);
+
+		// configure the maximum url length for SPARQL query GET requests
+		int maximumUrlLength = DEFAULT_MAXIMUM_URL_LENGTH;
+		String propertyValue = System.getProperty(MAXIMUM_URL_LENGTH_PARAM);
+		if (propertyValue != null) {
+			try {
+				maximumUrlLength = Integer.parseInt(propertyValue);
+			}
+			catch (NumberFormatException e) {
+				throw new RDF4JException("integer value expected for property " + MAXIMUM_URL_LENGTH_PARAM,
+						e)
+				{
+
+					private static final long serialVersionUID = 1L;
```

Comment:
The more I think on it, the more I like it: should we make RDF4JException concrete so we can just use it directly instead of always having to subclass? 

Given that it's a new class in RDF4J and not yet released, I can even get away with just fixing that as part of this PR :)